lib/install/management/appliance.go

```diff
@@ -634,18 +635,31 @@ func (d *Dispatcher) ensureComponentsInitialize(conf *config.VirtualContainerHos
 	}
 
 	dockerInfoURL := fmt.Sprintf("%s://%s:%s/info", proto, d.HostIP, d.DockerPort)
-	maxRetries := 10
-	for i := 0; i < maxRetries; i++ {
-		res, err = client.Get(dockerInfoURL)
+	req, err = http.NewRequest("GET", dockerInfoURL, nil)
+	if err != nil {
+		return errors.New("invalid HTTP request for docker info")
+	}
+	req = req.WithContext(d.ctx)
+
+	ticker := time.NewTicker(time.Second)
+	for {
+		res, err = client.Do(req)
 		if err == nil && res.StatusCode == http.StatusOK {
 			if isPortLayerRunning(res) {
-				return nil
+				break
```

Comment:
So basically here, we cannot query out error message from docker personality API. 
If there is anything wrong to initialize those services, user can only wait until timeout, without useful error message.